### PR TITLE
Clean up Terraform part 2

### LINKED ISSUE #103: Fix terraform variables - step 2

### PR DESCRIPTION
**Pull request by TessFerrandez**
- Clean up terraform naming, variables, and formatting
- Add script for managed identities
- Document terraform variables for step 2

Closes #103

### CODE REVIEW

File: 2-provision-aks/aks/aks.tf [hunk 1]
```diff
@@ -1,38 +1,37 @@
-
-resource "azurerm_kubernetes_cluster" "k8s" {
-    name                = var.cluster_name
+resource "azurerm_kubernetes_cluster" "k8s_cluster" {
+    name                = "aks-${var.cluster_name}"
     location            = var.location
     resource_group_name = var.resource_group_name
-    dns_prefix          = var.dns_prefix
+    dns_prefix          = var.cluster_name
```
Comment (Lybecker) [inline 2-provision-aks/aks/aks.tf:5]: The dns_prefix seemed odd, but it is a paramenter of the [Terraform K8s module](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster)

File: setup.md [hunk 2]
```diff
@@ -192,38 +192,54 @@ Before we provision the AKS clusters, we will provision some common resources th
...
     ```bash
     eval ACR_NAME=<acr_name>
     az acr build -t $ACR_NAME.azurecr.io/grafana:v1 -r $ACR_NAME .
     ```
 
-2. Create an identity for the cluster
+2. Create a managed identity for the cluster
 
-    MagicAKS creates a managed identity cluster. We create the identity for this cluster in the resource group with other long lasting resources, so the permissions remain even if we recreate the cluster. To create an identity follow the steps below:
+    MagicAKS creates a managed identity cluster. We create the identity for this cluster in the resource group with other shared resources, so the permissions remain even if we recreate the cluster. To create an identity run the `create-sluster-managed-identity.sh` script, providing the **resource_group_name** you entered in the terraform variables:
```
Comment (sachinkundu) [inline setup.md:212]: sluster -> cluster

Comment (sachinkundu) [reply setup.md:212]: you can even link to the file.

File: utils/scripts/create-cluster-managed-identity.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+if [ -z "$SHARED_RESOURCE_GROUP" ]
+then
+    echo "MagicAKS managed identity script"
+    echo
+    echo "Usage: $0 <shared resource group name>"
+    exit 1
+fi
+
+eval RG_WHERE_NETWORK_EXISTS=rg-magicaks-shared
```
Comment (sachinkundu) [inline utils/scripts/create-cluster-managed-identity.sh:16]: RG_WHERE_NETWORK_EXISTS is not used I think anymore as you have a parameter

**Review by sachinkundu — COMMENTED** [review]
Large patch so I am assuming you did all the deployment steps to verify all names are consistent.

**Review by deniscep — APPROVED** [review]

File: setup.md [hunk 4]
```diff
@@ -192,38 +192,54 @@ Before we provision the AKS clusters, we will provision some common resources th
...
-    az role assignment create --role "Network Contributor" --assignee $MSI_CLIENT_ID -g $RG_WHERE_NETWORK_EXISTS
-    az role assignment create --role "Virtual Machine Contributor" --assignee $MSI_CLIENT_ID -g $RG_WHERE_NETWORK_EXISTS
-    echo "MSI_RESOURCE_ID: $MSI_RESOURCE_ID"
+    ./utils/scripts/create-cluster-managed-identity.sh rg-magicaks-shared
     ```
 
     > **Note:** MagicAKS is not creating a system assigned managed identity, due to current [limitations](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity#create-an-aks-cluster-with-managed-identities) of self-managed VNet and static IP address outside the MC_ resource group.
 
-    You will need to provide the managed identity `MSI_RESOURCE_ID` as a variable to Terraform when creating the cluster.
+    You will need to provide the `Managed Identity Resource ID` as a variable to Terraform when creating the cluster.
```
Comment (tompaana) [inline setup.md:220]:
```suggestion
    You will need to provide the **managed identity resource ID** (provided as output of the script) as a variable to Terraform when creating the cluster.
```

File: utils/scripts/create-cluster-managed-identity.sh (hunk 3 above)
Comment (TessFerrandez) [reply utils/scripts/create-cluster-managed-identity.sh:16]: good catch

File: setup.md (hunk 2 above)
Comment (TessFerrandez) [reply setup.md:212]: fixing

**Review by tompaana — APPROVED** [review]
Wow

**Review by Lybecker — COMMENTED** [review]
Looks good

**Pull request merged**